lib/src/components/progress/progress_circle.dart

```diff
@@ -0,0 +1,131 @@
+import 'dart:math' as math;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../zeta_flutter.dart';
+import 'progress.dart';
+
+/// Sizes for [ZetaProgressCircle]
+enum ZetaCircleSizes {
+  ///24 X 24
+  xs,
+
+  /// 36 X 36
+  s,
+
+  /// 40 x 40
+  m,
+
+  /// 48 X 48
+  l,
+
+  /// 64 X 64
+  xl
+}
+
+///Class definition for [ZetaProgressCircle]
+class ZetaProgressCircle extends ZetaProgress {
+  /// Constructor for [ZetaProgressCircle]
+  const ZetaProgressCircle({
+    super.key,
+    super.progress = 0,
+    this.size = ZetaCircleSizes.xl,
+    this.rounded = true,
+  });
+
+  ///Size of [ZetaProgressCircle]
+  final ZetaCircleSizes size;
+
+  /// Border Type for [ZetaWidgetBorder]
+  final bool rounded;
+
+  @override
+  State<ZetaProgressCircle> createState() => ZetaProgressCircleState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(EnumProperty<ZetaCircleSizes>('size', size))
+      ..add(DoubleProperty('progress', progress))
+      ..add(DiagnosticsProperty<bool>('rounded', rounded));
+  }
+}
+
+///Class definition for [ZetaProgressCircleState]
+class ZetaProgressCircleState extends ZetaProgressState<ZetaProgressCircle> {
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
```

Comment:
We need to wrap this in a fixed size box, as there are no constraints. This is causing Widgetbook to fail, and would be a paint using the component in an app.
Wrap AnimatedBuilder in constraints like this: 
```dart
 return ConstrainedBox(
      constraints: BoxConstraints.tight(_getSize()),
      child: AnimatedBuilder(
```